### PR TITLE
Add x/rollup query server

### LINKED ISSUE #291: Add x/rollup query server
Add a [query server](https://docs.cosmos.network/v0.50/build/building-modules/messages-and-queries#queries) to `x/rollup` and expose endpoints for retrieving the following values from the store:
- L1BlockInfo
- Params

### PR DESCRIPTION
**Pull request by natebeauregard**
Closes #291 

Add a [query server](https://docs.cosmos.network/v0.50/build/building-modules/messages-and-queries#queries) to x/rollup and expose endpoints for retrieving the following values from the store:

- L1BlockInfo
- Params

## Release Notes

- **New Features**
	- Introduced a new rollup module with enhanced configuration capabilities, including genesis state management and parameter validation.
	- Added gRPC query methods for retrieving Layer 1 block information and rollup parameters.
	- Enhanced test coverage for rollup functionality, ensuring robust validation of fee withdrawal transactions and parameter queries.
	- Implemented a new `GenesisState` message for managing the genesis state of the rollup module.

- **Bug Fixes**
	- Improved error handling in various methods related to fee withdrawals and parameter retrieval.

- **Chores**
	- Updated dependencies for better JSON handling and streamlined codebase.
	- Refined naming conventions for constants and methods to enhance clarity and maintainability.

### CODE REVIEW

File: proto/rollup/v1/query.proto [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "rollup/v1/rollup.proto";
+
+option go_package = "github.com/polymerdao/monomer/x/rollup/types";
+
+// Query defines all query endpoints for the rollup module.
+service Query {
+  // Params returns all rollup module parameters.
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
```
Comment (joshklop) [inline proto/rollup/v1/query.proto:14]: is this endpoint standard for other modules?

File: x/rollup/keeper/deposits.go [hunk 2]
```diff
@@ -22,21 +22,6 @@ import (
 	"github.com/samber/lo"
 )
 
-// setL1BlockInfo sets the L1 block info to the app state
-//
-// Persisted data conforms to optimism specs on L1 attributes:
-// https://github.com/ethereum-optimism/optimism/blob/develop/specs/deposits.md#l1-attributes-predeployed-contract
-func (k *Keeper) setL1BlockInfo(ctx sdk.Context, info types.L1BlockInfo) error { //nolint:gocritic
```
Comment (joshklop) [inline x/rollup/keeper/deposits.go:29]: Why is this moved to the other file and why is it exported?

**Review by joshklop — APPROVED** [review]

File: proto/rollup/v1/query.proto (hunk 1 above)
Comment (natebeauregard) [reply proto/rollup/v1/query.proto:14]: Yep it follows the same format, here's the `Params` [endpoint](https://github.com/cosmos/cosmos-sdk/blob/99b4858a958c4986fe5eed367446c41b91a20ef6/x/bank/proto/cosmos/bank/v1beta1/query.proto#L71-L75) from `x/bank` as an example:

```
  // Params queries the parameters of x/bank module.
  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
    option (cosmos.query.v1.module_query_safe) = true;
    option (google.api.http).get               = "/cosmos/bank/v1beta1/params";
  }
```

File: x/rollup/keeper/deposits.go (hunk 2 above)
Comment (natebeauregard) [reply x/rollup/keeper/deposits.go:29]: I decided to group it with the setter/getter for `Params` and the getter for `L1BlockInfo` directly inside of `keeper.go` to abstract away the store accessing logic in `deposits.go`. This follows the Cosmos SDK convention of avoiding direct store accessing in message/query handling logic.

I made it exported to allow it to be tested in `grpc_query_test.go`. If you don't think it (or any of the other store accessors) should be exported I can either remove the unit test file or include it in the `keeper` package instead of `keeper_test`. Lmk your thoughts on that.
